src/Datadog.Trace/TracerSubProcessManager.cs

```diff
@@ -66,7 +101,7 @@ private static bool ProgramIsRunning(string fullPath)
             return false;
         }
 
-        private static Task StartProcessWithKeepAlive(string path, string args)
+        private static Task StartProcessWithKeepAlive(string path, string args, Action<Process> setProcess, Func<Process> getProcess)
```

Comment:
I think you can simplify this parameter list (especially the slightly weird get/set expressions) by passing the `SubProcessMetadata` as the only argument, like you do in `SafelyKillProcess()`.